### PR TITLE
lorem: add missing Python dependency

### PR DESCRIPTION
**Pull request by woodruffw**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Noticed during rebottling: `test` was failing for this formula, due to `python` not being present. Looks like the formula is a single-file Python program, so I've added the latest Python (`@3.12`) as a dependency.

### CODE REVIEW

File: Formula/l/lorem.rb [hunk 1]
```diff
@@ -10,7 +10,10 @@ class Lorem < Formula
     sha256 cellar: :any_skip_relocation, all: "79e38b6949ebd14157f0177fae6331e780f1889c76594c86dea7d649ef5c9057"
   end
 
+  depends_on "python@3.12"
+
   def install
+    inreplace "lorem", "/usr/bin/env python", which("python3.12")
     bin.install "lorem"
```
Comment (carlocab) [inline Formula/l/lorem.rb:17]:
```suggestion
    bin.install "lorem"
    rewrite_shebang detected_python_shebang, bin/"lorem"
```
You might need
```ruby
include Language::Python::Shebang
```

File: Formula/l/lorem.rb [hunk 2]
```diff
@@ -10,7 +10,10 @@ class Lorem < Formula
     sha256 cellar: :any_skip_relocation, all: "79e38b6949ebd14157f0177fae6331e780f1889c76594c86dea7d649ef5c9057"
   end
 
+  depends_on "python@3.12"
```
[the remark this answers is not in the document]
Comment (woodruffw) [reply Formula/l/lorem.rb:13]: It fails on Linux as well (ex: https://github.com/Homebrew/homebrew-core/actions/runs/10137061352/job/28026761459) -- does `uses_from_macos` work there as well?

[the remark this answers is not in the document]
Comment (branchvincent) [reply Formula/l/lorem.rb:13]: yup it's equivalent to:

```ruby
on_linux
    depends_on "python@3.12"
end
```

[the remark this answers is not in the document]
Comment (woodruffw) [reply Formula/l/lorem.rb:15]: TIL, thanks!

File: Formula/l/lorem.rb [hunk 3]
```diff
@@ -10,8 +12,11 @@ class Lorem < Formula
     sha256 cellar: :any_skip_relocation, all: "79e38b6949ebd14157f0177fae6331e780f1889c76594c86dea7d649ef5c9057"
   end
 
+  uses_from_macos "python"
+
   def install
     bin.install "lorem"
+    rewrite_shebang detected_python_shebang, bin/"lorem"
```
Comment (branchvincent) [inline Formula/l/lorem.rb:19]: this needs a slight tweak: https://github.com/Homebrew/homebrew-core/pull/178828#discussion_r1695367357

File: Formula/l/lorem.rb (hunk 1 above)
Comment (woodruffw) [reply Formula/l/lorem.rb]: Thanks!

**Review by branchv — APPROVED** [review]
looks good!

**Pull request merged**